server/db-middleware/task-api.js

```diff
@@ -31,4 +31,17 @@ apiModule.assignTask = function (assignmentInfo) {
   return newAssignment.save();
 };
 
+apiModule.deleteTask = function (taskId) {
+  return TaskAssignment
+    .find({ taskId })
+    .where('this.deadline > new Date().getMilliseconds()')
```

Comment:
getMilliseconds() will return current milliseconds count. So, It's number between 0 and 1000. Is it really what you want?

Reply:
Method where() is used for building queries. Read [it](http://mongoosejs.com/docs/api.html#query_Query-where). To use JavaScript expression in mongo query you need method $where(). Read [it](http://mongoosejs.com/docs/api.html#query_Query-$where). But $where operator is slow. Read [it](https://docs.mongodb.com/manual/reference/operator/query/where/). So, you should avoid it, if it's possible. And I think, that find() and operator $gt is enough for you.